helm-charts/splunk-otel-collector/templates/_helpers.tpl

```diff
@@ -441,6 +441,26 @@ Whether clusterReceiver should be enabled
 {{- end -}}
 
 
+{{/*
+Whether persistentQueue should be enabled
+*/}}
+{{- define "splunk-otel-collector.persistentQueueEnabledLogs" -}}
+{{ $gatewayDisabled := ne (include "splunk-otel-collector.gatewayEnabled" .) "true" }}
+{{- and $gatewayDisabled (.Values.splunkPlatform.sendingQueue.persistentQueueEnabled.logs) (and (ne (include "splunk-otel-collector.distribution" .) "eks/fargate") (ne (include "splunk-otel-collector.distribution" .) "gke/autopilot")) -}}
+{{- end -}}
```

Comment:
It looks like we only enable log persistence queue if gateway is disabled and we don't run in eks/fargate or autopilot, on top of our boolean field.
Can we document that limitation? Is there a reason we cannot do that on those platforms? Can we instead abort if someone tries to set that boolean to true?

Reply:
I mentioned the reasons in PR description.

Reply:
Pasting the same here:
```
Don't enable the persistent queues for following scenarios:

If the gateway mode is enabled. Gateway uses multiple replicas by default, and our filestorage extension will try to acquire an exclusive lock for the directory and it will cause a deadlock. Only one pod will be in a healthy state if we enable a persistent queue for gateway.
If the gke/autopilot or eks/fargate are used. Both of these distributions don't allow hostPath mounts.

```

Reply:
We should add both documention as well as validation rules for these limitations to ensure the best user experience. Please add validation rules with proper error messages to the values.schema.json file or in the the helm templates using the `fail` function.